packages/grafana-data/src/types/datasource.ts

```diff
@@ -413,6 +413,8 @@ export type ExploreQueryFieldProps<
 
 export interface QueryEditorHelpProps<TQuery extends DataQuery = DataQuery> {
   datasource: DataSourceApi<TQuery>;
+  queries: TQuery[];
+  refId: string;
```

Comment:
it sounds to me that you can pass only the `query` rather than the list of queries and the refId so you always need to look for it, isn't it?

Also, to make this change backwards compatible, you can make the `query` property optional.

Reply:
You're absolutely right! I was tracing the queries and forgot we also have query available. This cleans up things quite nicely, thank you.